File: openapi/code_samples/javascript/api_users_id.js
const express = require('express');
const app = express();
const port = 3000;

// Simulated user data for demonstration purposes
const users = [
    { id: 1, username: 'sampleuser1' },
    { id: 2, username: 'sampleuser2' },
    { id: 3, username: 'sampleuser3' }
];

app.get('/api/user/:userId', (req, res) => {
    const userId = parseInt(req.params.userId);
    const user = users.find(u => u.id === userId);

    if (user) {
        res.status(200).json(user);
    } else {
        res.status(400).json({ ErrorMessage: 'User not found.' });
    }
});

app.listen(port, () => {
    console.log(`Server is listening at http://localhost:${port}`);
});

File: openapi/code_samples/java/api_users_id.java
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/user")
public class UserController {

    // Simulated user data for demonstration purposes
    private final List<User> users = Arrays.asList(
            new User(1, "sampleuser1"),
            new User(2, "sampleuser2"),
            new User(3, "sampleuser3")
    );

    @GetMapping("/{userId}")
    public ResponseEntity<Object> findUserById(@PathVariable int userId) {
        User user = users.stream().filter(u -> u.getId() == userId).findFirst().orElse(null);
        if (user != null) {
            return ResponseEntity.ok(user);
        }

        return ResponseEntity.badRequest().body(new Fault("User not found."));
    }
}

class User {
    private int id;
    private String username;

    public User(int id, String username) {
        this.id = id;
        this.username = username;
    }

    // getters and setters
}
